bin/pelagos.py

```diff
@@ -0,0 +1,187 @@
+    app.logger.info('Return version')
+    return jsonify({'app': app.ver_name,
+                    'version': app.version})
+
+
+@app.route('/nodes', methods=['GET'])
+def get_nodes():
+    app.logger.info('Get all nodes information')
+    nodes = network_manager.load_data_file()
+    return jsonify(
+        {'nodes': network_manager.load_data_file(),
+            'pxe': pxelinux_cfg.get_pxe_map(nodes)})
+
+
+@app.route('/node/<string:node_id>', methods=['GET'])
+def get_node(node_id):
+    node = _check_input_node(node_id)
+    os = pxelinux_cfg.get_configured_os(node)
+    return jsonify({'node': node,
+                    'pxe': {'os': os,
+                            'pxe_file': pxelinux_cfg.get_macfile(node)}})
+
+
+@app.route('/check_image/<string:os>', methods=['GET'])
+def check_image(os):
+    os_id = _check_input_os(os)
+    return jsonify({'os': os_id})
+
+#
+# @app.route('/node/refresh_menu', methods=['GET'])
+# def refresh_menu():
+#    app.logger.info('Refresh menu. TBI')
+
+# PUT
+@app.route('/node/bootrecord/<string:node_id>/<string:os>', methods=['GET'])
+def bootrecord_node(node_id, os):
+    app.logger.info('Set boot record for node {}  with OS {}'.format(node_id, os))
+    node = _check_input_node(node_id)
+    os_id = _check_input_os(os)
+    pxelinux_cfg.set_tftp_dir(node, os_id)
+    return jsonify({'status': 'single boot record set',
+                    'node': node,
+                    'os': os_id
+                    })
+
+# DEL
+@app.route('/node/rmbootrecord/<string:node_id>', methods=['GET'])
+def rmbootrecord_node(node_id):
+    app.logger.info('Remove boot record for node {}'.format(node_id))
+    node = _check_input_node(node_id)
+    pxelinux_cfg.cleanup_tftp_dir(node)
+    return jsonify({'status': 'dedicated boot record removed',
+                    'node': node
+                    })
+
+#PATCH
+# os - PATCH parameter
+# node_id - parameter from url
+@app.route('/node/provision', methods=['POST'])
+@async_task
+def provision_node():
+    if request.method != 'POST':
+        return abort(405, "Use POST method")
+    # time.sleep(30)
+    app.logger.info("Data for provision: [" + str(request.get_data(as_text=True)) + "]")
+    os = request.form['os']
+    node_id = request.form['node']
+    app.logger.info(
+        'Provision node [{}]  with OS [{}]'.format(node_id, os))
+    node = _check_input_node(node_id)
+    os_id = _check_input_os(os)
+    if app.simulate_mode == 'fast':
+        pxelinux_cfg.provision_node_simulate_fast(node,os)
+    elif app.simulate_mode == 'medium':
+        pxelinux_cfg.provision_node_simulate(node, os)
+    else:
+        pxelinux_cfg.provision_node(node, os)
+    return jsonify({'status': 'done',
+                    'node': node,
+                    'os': os_id})
+
+
+# process command line parameters
+def print_help():
+    print('pelagos.py -c <config file>')
+
+
+if __name__ == '__main__':
+    try:
+
+        opts, args = getopt.getopt(sys.argv[1:],
```

Comment:
nit: maybe it is better to use argparse or docopt?

Reply:
done